app/packages/app/src/components/DatasetSelector.tsx

```diff
@@ -19,14 +24,19 @@ const DatasetSelector: React.FC<{
 }> = ({ useSearch }) => {
   const setDataset = useSetDataset();
   const dataset = useRecoilValue(datasetName) as string;
+  const resetFieldVisibility = useResetRecoilState(fieldVisibilityStage);
+
   return (
     <Selector<string>
       component={DatasetLink}
       placeholder={"Select dataset"}
       inputStyle={{ height: 40, maxWidth: 300 }}
       containerStyle={{ position: "relative" }}
       onSelect={(name) => {
-        name !== dataset && setDataset(name);
+        if (name !== dataset) {
+          setDataset(name);
+          resetFieldVisibility();
```

Comment:
We definitely want field visibility to reset on dataset change, but I think that should be captured in the page transition via `subscribe` in `@fiftyone/relay`. Can discuss offline